### PR TITLE
[KFP] Switch workflow runner to dedicated mlrun-kfp image

### PR DESCRIPTION
**Pull request by adamdelman**
[KFP] Switch workflow runner to dedicated mlrun-kfp image

This pull request builds upon the restructured pipeline adapter dependencies introduced in PR #6918.

It updates the pipeline workflow runner to utilize the new mlrun-kfp image, ensuring compatibility with the latest enhancements in the Kubeflow Pipelines (KFP) ecosystem.

### CODE REVIEW

File: pipeline-adapters/mlrun-pipelines-kfp-common/setup.py [hunk 1]
```diff
@@ -38,6 +38,11 @@
         "mlrun",
         "kfp",
     ],
+    install_requires=[
+        "graphviz~=0.20.0",
```
Comment (liranbg) [inline pipeline-adapters/mlrun-pipelines-kfp-common/setup.py]: isnt `graphviz` and `ipython` is only for when installing on jupyter? or so?
perhaps not needed when installing on mlrun/mlrun-kfp images?

File: pipeline-adapters/mlrun-pipelines-kfp-common/setup.py [hunk 2]
```diff
@@ -38,6 +38,11 @@
...
+        "kubernetes~=25.3.0",
```
Comment (liranbg) [inline pipeline-adapters/mlrun-pipelines-kfp-common/setup.py]: I wouldnt lock the `kubernetes` SDK with ~ but rather >=. any reason it is 25.3 and not newer one?

File: tests/test_requirements.py [hunk 3]
```diff
@@ -146,6 +152,7 @@ def test_requirement_specifiers_convention():
         "scikit-learn": {"~=1.5.1"},
         # ensure minimal version to gain vulnerability fixes
         "setuptools": {">=75.2"},
+        "mlrun_pipelines_kfp_v2": {">=0.2.5 ; python_version >= '3.11'"},
```
Comment (liranbg) [inline tests/test_requirements.py:155]: >= or <=?

File: dockerfiles/mlrun-kfp/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+ARG MLRUN_DOCKER_REGISTRY
+ARG MLRUN_VERSION=unstable
+# Compute the base image name, adding a slash only if the registry is not empty
+ARG BASE_IMAGE=${MLRUN_DOCKER_REGISTRY:+${MLRUN_DOCKER_REGISTRY}/}mlrun/mlrun
+
+FROM ${BASE_IMAGE}:${MLRUN_VERSION}
+
+RUN python -m pip install .[kfp18]
```
Comment (liranbg) [inline dockerfiles/mlrun-kfp/Dockerfile]: can install using uv? see mlrun/mlrun or mlrun/api image how it is done

**Review by liranbg — COMMENTED** [review]
minor

File: pipeline-adapters/mlrun-pipelines-kfp-common/setup.py (hunk 1 above)
Comment (adamdelman) [reply pipeline-adapters/mlrun-pipelines-kfp-common/setup.py]: fixed

File: pipeline-adapters/mlrun-pipelines-kfp-common/setup.py (hunk 2 above)
Comment (adamdelman) [reply pipeline-adapters/mlrun-pipelines-kfp-common/setup.py]: newer version have a conflict, don't remember what it was exactly

File: tests/test_requirements.py (hunk 3 above)
Comment (adamdelman) [reply tests/test_requirements.py:155]: Which part are you referring to?

File: dockerfiles/mlrun-kfp/Dockerfile (hunk 4 above)
Comment (adamdelman) [reply dockerfiles/mlrun-kfp/Dockerfile]: fixed

File: dependencies.py [hunk 5]
```diff
@@ -87,6 +87,8 @@ def extra_requirements() -> dict[str, list[str]]:
     )
     extras_require.update(
         {
+            "kfp18": ["mlrun_pipelines_kfp_v1_8[kfp]>=0.2.5 ; python_version < '3.11'"],
+            "kfp2": ["mlrun_pipelines_kfp_v2[kfp]>=0.2.5 ; python_version >= '3.11'"],
```
Comment (liranbg) [inline dependencies.py]:
```suggestion
            # TODO: uncomment when kfp 2 client is GAed
            #"kfp2": ["mlrun_pipelines_kfp_v2[kfp]>=0.2.5 ; python_version >= '3.11'"],
```

**Review by liranbg — APPROVED** [review]
** https://github.com/mlrun/mlrun/pull/6918#pullrequestreview-2508990125 is answered and its changes are reflected here as well

File: dependencies.py (hunk 5 above)
Comment (adamdelman) [reply dependencies.py]: fixed

**Pull request merged**